tests/nat/nat_helpers.py

```diff
@@ -854,23 +853,33 @@ def generate_and_verify_icmp_traffic(ptfadapter, setup_info, interface_type, dir
     # Create packet to send
     pkt = create_packet(network_data.eth_dst, network_data.eth_src, network_data.ip_dst, network_data.ip_src, protocol_type)
     # Define expected packet(ICMP request)
-    exp_pkt_request = expected_mask_nated_packet(pkt, protocol_type, network_data.exp_dst_ip, network_data.exp_src_ip, icmp_id=icmp_id)
+    exp_pkt_request = create_packet(network_data.eth_dst, network_data.eth_src, network_data.exp_dst_ip, network_data.exp_src_ip, protocol_type)
     # Reverse source and destination IPs for reply
     exp_dst_ip = get_src_ip(setup_info, direction, interface_type,
                             nat_type=nat_type, second_port=second_port)
     exp_src_ip = get_dst_ip(setup_info, direction, interface_type,
                             nat_type=nat_type)
     # Define expected packet(ICMP reply)
-    exp_pkt_reply = expected_mask_nated_packet(pkt, protocol_type, exp_dst_ip, exp_src_ip, icmp_id=0)
+    exp_pkt_reply = expected_mask_nated_packet(pkt, protocol_type, exp_dst_ip, exp_src_ip)
     exp_pkt_reply.exp_pkt[protocol_type].type = 0
+    exp_pkt_reply.exp_pkt[protocol_type].id = 0
     # clear buffer
     ptfadapter.dataplane.flush()
     # Send packet
     for port in network_data.inner_ports:
         testutils.send(ptfadapter, port, pkt, count=5)
     # Verify ICMP request packets arrive on outer ports
-    testutils.verify_packet_any_port(ptfadapter, exp_pkt_request, ports=network_data.outer_ports)
-    # Verify ICMP peply packets arrive on inner ports
+    pkt_filter = FilterPktBuffer(ptfadapter=ptfadapter,
+                                 exp_pkt=exp_pkt_request,
+                                 dst_port_numbers=network_data.outer_ports,
+                                 match_fields=[("IP", "src"), ("IP", "dst")],
+                                 ignore_fields=[("Ether", "src"), ("Ether", "dst"), ("IP", "chksum"), ("IP", "ttl")])
+    pkt_filter.filter_pkt_in_buffer()
+    pkt_dict = convert_pkt_to_dict(pkt_filter.received_pkt)
+    # Verify icmp_id in arrived packet
+    if icmp_id != None:
+        pytest_assert(int(pkt_dict['ICMP']['id']) in range(icmp_id, icmp_id+1001), "icmp_id: {} not in pool range".format(pkt_dict['ICMP']['id']))
```

Comment:
IMHO magic number is bad practice, I'd recommend refactoring `generate_and_verify_icmp_traffic` arguments to take icmp_id_range(use tuple with start and end value or ethier separate arg for end value)

Reply:
Hi @msosyak, I refactored `generate_and_verify_icmp_traffic` by adding separate argument for end value